lib/phoenix_html/link.ex

```diff
@@ -45,20 +45,23 @@ defmodule Phoenix.HTML.Link do
   end
 
   def link(text, opts) do
-    {to, opts} = Keyword.pop(opts, :to)
-    {method, opts} = Keyword.pop(opts, :method, :get)
-
-    unless to do
-      raise ArgumentError, "option :to is required in link/2"
-    end
-
-    if method == :get do
-      content_tag(:a, text, [href: to] ++ opts)
-    else
-      {form, opts} = form_options(opts, method, "link")
-      form_tag(to, form) do
-        content_tag(:a, text, [href: "#", onclick: "this.parentNode.submit(); return false;"] ++ opts)
-      end
+    cond do
+      not match?([{_key, _value}|_opts], opts) ->
```

Comment:
Wouldn't it be better to have a `not is_list/1` guard as another function clause above the current one?